pkg/orchestrator/scheduler/batch_service_job.go

```diff
@@ -105,52 +105,80 @@ func (b *BatchServiceJobScheduler) Process(ctx context.Context, evaluation *mode
 	}()
 
 	ctx = log.Ctx(ctx).With().Str("JobID", evaluation.JobID).Str("EvalID", evaluation.ID).Logger().WithContext(ctx)
-
-	job, jobExecutions, err := b.loadJobState(ctx, metrics, evaluation)
+	job, previouslyCreatedExecutionsList, err := b.loadJobState(ctx, metrics, evaluation)
 	if err != nil {
 		return err
 	}
 
 	// Plan to hold the actions to be taken
 	plan := models.NewPlan(evaluation, &job)
-	existingExecs := execSetFromSliceOfValues(jobExecutions)
-	nonTerminalExecs := existingExecs.filterNonTerminal()
+	allPreviouslyCreatedExecutionsSetRaw := execSetFromSliceOfValues(previouslyCreatedExecutionsList)
+
+	currentRuntimeIDExecutionsSet := execSet{}
+	otherRuntimeIDExecutionsSet := execSet{} // These executions will be eventually marked for cancellation
+
+	if evaluation.RuntimeID != "" {
+		// A runtime ID was passed in the evaluation, make
+		currentRuntimeIDExecutionsSet = allPreviouslyCreatedExecutionsSetRaw.filterByRuntimeID(evaluation.RuntimeID)
+		otherRuntimeIDExecutionsSet = allPreviouslyCreatedExecutionsSetRaw.excludeThisRuntimeID(evaluation.RuntimeID)
+	} else {
+		// In case no runtime ID was passed in the evaluation
+		currentRuntimeIDExecutionsSet = allPreviouslyCreatedExecutionsSetRaw
+	}
+
+	nonTerminalCurrentRuntimeIDExecutionsSet := currentRuntimeIDExecutionsSet.filterNonTerminal()
 
-	// early exit if the job is stopped
-	if job.IsTerminal() {
-		nonTerminalExecs.markCancelled(plan, orchestrator.ExecStoppedByJobStopEvent())
+	// If the job is already in a terminal state (Completed/Failed/Stopped),
+	// cancel any existing executions and exit
+	if job.IsTerminal() && evaluation.TriggeredBy != models.EvalTriggerJobRerun {
```

Comment:
Not related to this, but we need to figure out the housekeeper, which rlies on the Job creation time, which we are not changing when updating, we only change the modify time

Reply:
Do not rely on the evalution trigger

Reply:
Not applicable anymore